src/_pytest/pytester.py

```diff
@@ -1086,8 +1099,15 @@ def run(self, *cmdargs, **kwargs):
         try:
             now = time.time()
             popen = self.popen(
-                cmdargs, stdout=f1, stderr=f2, close_fds=(sys.platform != "win32")
+                cmdargs,
+                stdout=f1,
+                stderr=f2,
+                close_fds=(sys.platform != "win32"),
+                **popen_kwargs
             )
+            if stdin is not CLOSE_STDIN:
+                popen.stdin.write(stdin)
+                popen.stdin.close()
```

Comment:
This is a bit wonky here (closing with "not CLOSE_STDIN"), but it is due to compatibility with `popen`'s behavior mostly.

Reply:
Improved by handling `bytes` in a special way.